packages/framework-core/src/services/read-model-store.ts

```diff
@@ -32,48 +35,26 @@ export class ReadModelStore {
       return
     }
 
-    await Promise.all(
+    await Promises.allSettledAndFulfilled(
       projections.map(async (projectionMetadata: ProjectionMetadata) => {
         const readModelName = projectionMetadata.class.name
         const entitySnapshot = entitySnapshotEnvelope.value as EntityInterface
         const readModelID = this.joinKeyForProjection(entitySnapshot, projectionMetadata)
-        const readModel = await this.fetchReadModel(readModelName, readModelID)
         this.logger.debug(
           '[ReadModelStore#project] Projecting entity snapshot ',
           entitySnapshotEnvelope,
-          ' to build new state of read model ${readModelName} with ID ${readModelID}'
+          ` to build new state of read model ${readModelName} with ID ${readModelID}`
         )
-        const newReadModel = this.projectionFunction(projectionMetadata)(entitySnapshot, readModel)
 
-        if (newReadModel === ReadModelAction.Delete) {
-          this.logger.debug(
-            `[ReadModelDelete#project] Deleting read model ${readModelName} with ID ${readModelID}:`,
-            readModel
-          )
-          return this.provider.readModels.delete(this.config, this.logger, readModelName, readModel)
-        } else if (newReadModel === ReadModelAction.Nothing) {
-          this.logger.debug(
-            `[ReadModelStore#project] Skipping actions for ${readModelName} with ID ${readModelID}:`,
-            newReadModel
-          )
-          return
-        }
-        this.logger.debug(
-          `[ReadModelStore#project] Storing new version of read model ${readModelName} with ID ${readModelID}:`,
-          newReadModel
+        return await retryIfError(
+          this.logger,
+          () => this.applyProjectionToReadModel(readModelName, readModelID, projectionMetadata, entitySnapshot),
+          OptimisticConcurrencyUnexpectedVersionError
```

Comment:
The above logic (read the read model, update, and store it) was moved to a method. Now we use the `retryIfError` function here. The logic will be retried if the provider throws `OptimisticConcurrencyUnexpectedVersionError`